torch/_inductor/codecache.py

```diff
@@ -1648,14 +1650,28 @@ def _pad_to_alignment(raw_bytes: bytes) -> bytes:
             if config.aot_inductor.force_mmap_weights:
                 use_mmap_weights = True
 
+            # precompile the AOT header for this device
+            header_file = _get_cpp_wrapper_header(device_type, aot_mode=graph.aot_mode)
+            compile_command = {
+                "vec_isa": picked_vec_isa,
+                "device_type": device_type,
+                "aot_mode": graph.aot_mode,
+                "use_absolute_path": use_absolute_path,
+                "use_mmap_weights": use_mmap_weights,
+            }
             object_build_options = CppTorchDeviceOptions(
-                vec_isa=picked_vec_isa,
-                device_type=device_type,
-                aot_mode=graph.aot_mode,
                 compile_only=True,
-                use_absolute_path=use_absolute_path,
-                use_mmap_weights=use_mmap_weights,
+                **compile_command,  # type: ignore[arg-type]
             )
+            object_build_options.set_precompiled_header(
+                _precompile_header(
```

Comment:
Let's separate this for cpp-wrapper and AOTI. I would like to see this rolling out for cpp-wrapper first, and then later applying to AOTI. Also it's worth to add an inductor config to control this behavior, in case caching goes bad and we can easily triage.

Reply:
@desertfire Would you prefer separate commits, or would separate config variables do the trick?

Reply:
I went with config variables, unless you have objections.